engine/runtime/src/main/scala/org/enso/compiler/phase/ImportResolver.scala

```diff
@@ -133,11 +134,13 @@ class ImportResolver(compiler: Compiler) {
   }
 
   private def tryResolveAsType(
-    name: IR.Name.Qualified
+    name: IR.Name.Qualified,
+    bindingsCachingEnabled: Boolean
   ): Option[ResolvedType] = {
     val tp  = name.parts.last.name
     val mod = name.parts.dropRight(1).map(_.name).mkString(".")
     compiler.getModule(mod).flatMap { mod =>
+      mapImports(mod, bindingsCachingEnabled)
```

Comment:
As a side-effect this populates `module.bindings` from cache, if there are there. That prevents a `compiler.ensureParsed` to create "empty" `BindingsMap` without any parsed symbols.

Reply:
I don't think this is correct. This will easily trigger SO when there are mutual dependencies between the module and the module that is imported.